### PR TITLE
Fixed #35991 -- Fixed crash when adding non-nullable field after renaming part of CompositePrimaryKey on SQLite.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-35991

### CODE REVIEW

File: django/db/migrations/state.py [hunk 1]
```diff
@@ -323,6 +323,16 @@ def rename_field(self, app_label, model_name, old_name, new_name):
                         for from_field_name in from_fields
                     ]
                 )
+            # Fix field names (e.g. for CompositePrimaryKey) to refer to the
+            # new field.
+            if field_names := getattr(field, "field_names", None):
```
Comment (charettes) [inline django/db/migrations/state.py:328]: I guess we'll want to eventually have a `composite` flag on `CompositeField` (e.g. like we have with `many_to_many` and friends) but that's the best we can do right now.

**Review by charettes — APPROVED** [review]
Thanks for the patch @felixxm, hope you'll have nice hollidays!

**Comment by felixxm** [comment]
> Thanks for the patch @felixxm, hope you'll have nice hollidays!

Thanks for the review :+1: Happy Holidays! :christmas_tree: :gift:

**Review by nessita — APPROVED** [review]
Thank you Mariusz! This looks great :star2:

**Pull request merged**